projects/blenderbot2/agents/blenderbot2.py

```diff
@@ -507,7 +515,7 @@ def _set_query_vec(self, observation: Message) -> Message:
             )
         if self.add_person_tokens:
             query_str = self._remove_person_tokens(query_str)
-        observation['query_vec'] = self.model.tokenize_query(query_str)
+        observation['query_vec'] = self.model_api.tokenize_query(query_str)
```

Comment:
Shouldn't this be `self.model_api().tokenize_query(query_str)`?

Reply:
idts, model_api is a `@property`

Kurt, maybe we should bump this all the way up to TA. We have this `module` stuff everywhere

Reply:
Oh, yeah right. I missed that.

Reply:
> bump up to TA

We would need to think of an elegant solution so that we can still call `self.model()` directly (which _does_ work in distributed). We could solve it with `model_api` if you think that's appropriate

Reply:
you're right, nvm